cpp/src/arrow/array/builder_nested.h

```diff
@@ -40,37 +40,46 @@ namespace arrow {
 /// @{
 
 // ----------------------------------------------------------------------
-// List builder
+// VarLengthListLikeBuilder
 
 template <typename TYPE>
-class BaseListBuilder : public ArrayBuilder {
+class ARROW_EXPORT VarLengthListLikeBuilder : public ArrayBuilder {
```

Comment:
I would keep the `Base` prefix to stress that it's not the final implementation class, e.g. `BaseVarLengthListBuilder`.

Reply:
```
Base+VarLengthListLikeArray

or 

Base+VarLengthListArray
```

?

Reply:
Fair enough, let's not bikeshed this :-)